expect/_equal.ts

```diff
@@ -73,6 +73,13 @@ export function equal(c: unknown, d: unknown, options?: EqualOptions): boolean {
     if (typeof a === "number" && typeof b === "number") {
       return Number.isNaN(a) && Number.isNaN(b) || a === b;
     }
+    if (a === null || b === null) {
+      return a === b;
+    }
+    const className = Object.prototype.toString.call(a);
+    if (className != Object.prototype.toString.call(b)) {
```

Comment:
Nit: should the inequality be stricter?

Reply:
Good point, yeah that should've been a strict equality check.